numpy/ma/core.py

```diff
@@ -3183,29 +3185,80 @@ def __getitem__(self, indx):
         Return the item described by i, as a masked array.
 
         """
-        dout = self.data[indx]
         # We could directly use ndarray.__getitem__ on self.
         # But then we would have to modify __array_finalize__ to prevent the
         # mask of being reshaped if it hasn't been set up properly yet
         # So it's easier to stick to the current version
+        dout = self.data[indx]
         _mask = self._mask
+
+        def _is_scalar(m):
```

Comment:
> in cases where I want access to some of the local variables

Note that the original implementation did exactly that, but I realized it was pretty trivial to convert them into parameters, as there were only two